Python/ceval_macros.h

```diff
@@ -122,6 +140,24 @@ do { \
         DISPATCH_GOTO(); \
     }
 
+#ifdef Py_TAIL_CALL_INTERP
+#define DISPATCH_INLINED(NEW_FRAME) \
```

Comment:
I think we would improve clarity and maybe reduce code size of we got rid of `DISPATCH_INLINED` and replaced it with `START_FRAME(NEW_FRAME); DISPATCH()`.

`START_FRAME` can set up execution of the new frame (which is independent of the interpreter dispatch mechanism).

Reply:
Can we do that in another PR after this one? The reason is that DISPATCH_INLINED in a tail call does the following:

1. Start a new frame.
2. Enter new frame and do recursive check.
3. Lltrace
4. dispatch.

Whereas the normal start_frame in the switch-case does:

1. Start frame.
2. Dispatch.

So we can apply your suggestion for the base interpreter, but it wouldn't help much for this one.

Reply:
Both must do the same thing. The only difference is how the code is laid out.

What I'm suggesting is to change the layout the code such that the only difference is tailcall vs jump.

It might make sense to add a new `start_frame` label and jump to that.

Reply:
Adding a start_frame label will lower perf on the default build (two jumps instead of one). I propose a macro instead.

Reply:
I added a label for start_frame.